pkg/trait/jvm.go

```diff
@@ -81,10 +80,18 @@ func (t *jvmTrait) Configure(e *Environment) (bool, *TraitCondition, error) {
 	if e.CamelCatalog == nil {
 		return false, NewIntegrationConditionPlatformDisabledCatalogMissing(), nil
 	}
+
+	if t.Jar == "" && e.IntegrationKit != nil {
+		executableArtifact := e.IntegrationKit.GetExecutableArtifact()
+		if executableArtifact == nil {
+			return false, nil, fmt.Errorf("could not locate any executable among Integration Kit %s artifacts", e.IntegrationKit.Name)
+		}
+		t.Jar = executableArtifact.Target
```

Comment:
maybe a stupid question since I'm unable to figure out how all the bits are wiring together, but is this backward compatible ? i.e. what would happen for already existing kits & running integrations ?

Reply:
No, it's a pertinent question. I am still working on the backward compatibility. I was thinking to comment that part and forcefully add "dependencies/quarkus-run.jar" which should be kept until we support V2.3. As this part is also overlapping the development I am doing to support the trait for "sourceless" Integration is something I'll need to still do. Since it's still a draft PR, consider that certain details are missing. However, definetely good to have early feedback, thanks!

Reply:
unrelated, but it would be nice to have some design proposal for those _sourceless_ integration because as today, I don't have a complete understanding of how it is different from a normal integration (except from the fact that the jvm trait should probably not be required in that case)

Reply:
Sure, we can open a separate thread to discuss anything related to this. As you've said, technically speaking there is no difference between the *managed* Integrations. The only difference is that these Kits are not built by the operator, so, it cannot make most of the assumptions around certain aspects (above all linked to the catalog, see also #5417). All the documentation so far is on https://camel.apache.org/camel-k/next/running/camel-runtimes.html

BTW, I'm planning to add some more documentation as part of this PR to explain how the JVM changes can enhance the life of users.

Reply:
yeah, a separate discussion would be very useful